File: Dagger2Ex/app/src/main/java/expencive/vk/com/dagger2ex/dagger/DieselEngineModule.java
package expencive.vk.com.dagger2ex.dagger;


import dagger.Binds;
import dagger.Module;
import expencive.vk.com.dagger2ex.car.DieselEngine;
import expencive.vk.com.dagger2ex.car.Engine;

@Module
public abstract class DieselEngineModule {

    @Binds
    abstract Engine bindEngine(DieselEngine engine);

}

File: DaggerKotlin/app/src/main/java/vk/expencive/daggerkotlin/dagger/DieselEngineModule.kt
package vk.expencive.daggerkotlin.dagger

import dagger.Binds
import dagger.Module
import dagger.Provides
import vk.expencive.daggerkotlin.car.DieselEngine
import vk.expencive.daggerkotlin.car.Engine


@Module
class DieselEngineModule(private var horsePower: Int) {

    @Provides
    fun provideDieselEngine(): Engine = DieselEngine(horsePower)
}